ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala

```diff
@@ -414,28 +414,27 @@ object BigDiffy extends Command {
     sys.exit(1)
   }
 
-  private def avroKeyFn(key: String): GenericRecord => String = {
+  private def avroKeyFn(keys: Seq[String]): GenericRecord => String = {
     @tailrec
     def get(xs: Array[String], i: Int, r: GenericRecord): String =
       if (i == xs.length - 1) {
         r.get(xs(i)).toString
       } else {
         get(xs, i + 1, r.get(xs(i)).asInstanceOf[GenericRecord])
       }
-    val xs = key.split('.')
-    (r: GenericRecord) => get(xs, 0, r)
+    (r: GenericRecord) =>  keys.map { k => get(k.split('.'), 0, r) }.mkString("_")
```

Comment:
Hmm, what if we change this to a `Set[String]` instead? My concern is this can produce some weird edge cases. They will probably be rare but it's not really transparent to end user. 

e.g.
`key1_something, key2` vs `key1, something_key2` produces the same aggregate key.

Reply:
Only suggesting `Set` since we use that in sampler but open to other ideas

Reply:
wouldn't it be more confusing is the results don't respect the user ordering of the keys?

Reply:
Maybe some dataset that preserves order instead. Up to you. Doesn't block this PR imo but would be nice to have fixed :)